policy_definitions/tagging/policy_assignment_qby_deploy_require_tags.tmpl.json

```diff
@@ -0,0 +1,29 @@
+{
+    "name": "QBY-Require-Tags",
+    "type": "Microsoft.Authorization/policyAssignments",
+    "apiVersion": "2022-06-01",
+    "dependsOn": [],
+    "properties": {
+      "description": "Automatically generated policy set to enforce existence of a tag on resource groups and all resources.",
+      "displayName": "Require tags",
+      "enforcementMode": null,
+      "nonComplianceMessages": [
+        {
+          "message": "The resource group has not required tag",
+          "policyDefinitionReferenceId": "${tag_name}-ResourceGroups"
+        },
+        {
+          "message": "The resource has not required tag",
+          "policyDefinitionReferenceId": "${tag_name}-Resources"
+        }
```

Comment:
That won't work, as you don't use a template function to create the objects based on the tags.

Reply:
Fixed with the new custom policy definition.